src/main/java/com/google/api/codegen/discovery/transformer/SurfaceNamer.java

```diff
@@ -0,0 +1,31 @@
+/* Copyright 2016 Google Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.api.codegen.discovery.transformer;
+
+import com.google.api.codegen.util.NameFormatter;
+import com.google.api.codegen.util.NameFormatterDelegator;
+
+/**
+ * TODO(saicheems)
+ */
+public class SurfaceNamer extends NameFormatterDelegator {
```

Comment:
I recommend calling this DiscoverySampleNamer (or SampleNamer if you don't want 'discovery' in it).